posts/superseding-bpmodels/index.qmd

```diff
@@ -0,0 +1,49 @@
+
+One of the first considerations was the scope of the package. When maintaining or re-imagining an R package, assessing its scope and identifying opportunities for refinement is crucial. For example, some packages have evolved significantly in the broader R ecosystem to better align with user needs. We will highlight a few examples.[`{plyr}`](https://github.com/hadley/plyr) was split into [`{dplyr}`](https://dplyr.tidyverse.org/index.html) and [`{purrr}`](https://purrr.tidyverse.org/index.html) for manipulating data frame and list objects respectively, reflecting more specialized functionality based on object types. Similarly, `{reshape}` evolved into [`{reshape2}`](https://github.com/hadley/reshape)[^1] and currently into [`{tidyr}`](https://tidyr.tidyverse.org/index.html), with each iteration simplifying and improving upon its predecessor. Another example is the renaming[^2] of `{ggmissing}` into the more generalized [`{naniar}`](https://naniar.njtierney.com/). In the epidemiology ecosystem, two examples include the evolution of the [`{EpiNow}`](https://epiforecasts.github.io/EpiNow/) into [`{EpiNow2}`](https://epiforecasts.github.io/EpiNow2/) and [`{incidence}`](https://www.repidemicsconsortium.org/incidence/) to [`{incidence2}`](https://www.reconverse.org/incidence2/index.html).
+
+For {bpmodels}, we wanted to unify the simulation functions (existing as two functions previously) and improve the function signature by renaming several of the arguments for readability. We also wanted to introduce an object-oriented workflow to aid in interoperability with existing tools such as [epicontacts](https://www.repidemicsconsortium.org/epicontacts/) and [epiparameter](https://epiverse-trace.github.io/epiparameter/). The object-oriented backend would also allow us to implement better methods for printing, summarising and aggregating the simulation output. Some of these considerations would have been less disruptive than others but the change in function name and signature would have led to a lot of disruptions including deprecating the existing functions and arguments.
+
+One thing is clear from the examples on scope changes - they often lead to name changes. Another important decision was whether to rebrand the package with a new name. A new name can signal a fresh approach and address limitations of the original package. The most popular example is the renaming of `{ggplot}` to [`{ggplot2}`](https://ggplot2.tidyverse.org/index.html). Other examples include renaming {reshape} to {reshape2} and {tidyr}. In our case, we decided to fork the original [`{bpmodels}`](https://epiforecasts.github.io/bpmodels/) repository under Epiverse-TRACE and maintain the old package to avoid breaking scripts that rely on it. At the same time, we introduced {epichains} as the successor. The name reflects the fact that it is a package for analysing *epidemiological transmission chains*.
+
```

Comment:
```suggestion

A second key consideration is the plans that the original package author(s) may have had and their views on any future changes. In our case this was fairly straightforward because the maintainer of the original package was fully involved in the refactoring. Another package author who had made substantial contributions could be reached and approval sought. More generally, however, bringing all package authors on board with things like scope change, name change etc. is an important step in taking on maintenance of a package and one that must not be neglected.

```